frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -137,7 +149,11 @@ function NewRunParametersV2(props: NewRunParametersProps) {
                   nextUpdatedParameters[k] = { string_value: value };
                   setUpdatedParameters(nextUpdatedParameters);
                   if (props.handleParameterChange) {
-                    props.handleParameterChange(nextUpdatedParameters);
+                    let parametersInRealType = {};
+                    Object.entries(nextUpdatedParameters).map(([k, v]) => {
+                      parametersInRealType[k] = inputConverter(v['string_value']);
```

Comment:
We should get rid of `string_value` and directly store the string in `v`. For example: https://github.com/kubeflow/pipelines/blob/e6832909e8b774a3aef02481a24f0dac4ee387f8/frontend/src/components/NewRunParametersV2.tsx#L137

Reply:
The v here represents an object {string_value: userInputStr}, I am not sure if storing the whole object is better than only storing the userInputStr.

Reply:
To clarify: We shouldn't use `object` type for `v`, but should use `string` type for `v`. We don't need to store the object because all we need is just the string value that user types in, correct? Please let me know if I am missing anything.

Reply:
The v here is from nextUpdatedParameters which is in form of {key: {string_value: userInputStr}}.
That means that v = {string_value: userInputStr}, so we use v['string_value'] to retrieve the userInputStr
Why v in nextUpdatedParameters is an **object** rather than pure string? In order to match the props in TextField.

Reply:
Solved. Change v to pure string!